lib/assert.js

```diff
@@ -261,10 +264,8 @@ function getErrMessage(message, fn) {
   Error.captureStackTrace(err, fn);
   Error.stackTraceLimit = tmpLimit;
 
-  const tmpPrepare = Error.prepareStackTrace;
-  Error.prepareStackTrace = (_, stack) => stack;
+  internalOverriddenErrors.set(err, (_, stack) => stack);
```

Comment:
just a nit: the function name here doesn't really make it obvious that this is a replacement for `prepareStackTrace()`. It would also be good to include some documentation comments somewhere here that explains why `prepareStackTrace()` shouldn't be used.